app/Helpers/database/player-achievement.php

```diff
@@ -279,74 +279,82 @@ function getUnlocksInDateRange(array $achievementIDs, string $startTime, string
  */
 function getAchievementDistribution(
     int $gameID,
-    int $hardcore,
+    int $isHardcore,
     ?string $requestedBy = null,
     int $flag = AchievementFlag::OfficialCore,
     int $numPlayers = 0
 ): array {
     /** @var Game $game */
-    $game = Game::withCount(['achievements' => fn ($query) => $query->flag($flag)])
-        ->find($gameID);
+    $game = Game::withCount(['achievements' => fn ($query) => $query->flag($flag)])->find($gameID);
+    $results = null;
 
     if (!$game || !$game->achievements_count) {
         // NOTE this will return an empty array instead of an empty object. keep it like this for backwards compatibility.
         return [];
     }
 
-    $bindings = [
-        'gameId' => $gameID,
-    ];
-
     // if a game has more than 100 players, don't filter out the untracked users as the
     // join becomes very expensive. will be addressed when denormalized data is captured
-    $joinStatement = '';
-    $joinStatementUnofficial = '';
-    $requestedByStatement = '';
-    if ($numPlayers < 100) {
-        $joinStatement = 'INNER JOIN UserAccounts AS ua ON ua.ID = pg.user_id';
-        $joinStatementUnofficial = 'INNER JOIN UserAccounts AS ua ON ua.ID = pa.user_id';
-        $requestedByStatement = 'AND (ua.Untracked = 0';
-        if ($requestedBy) {
-            $bindings['requestedBy'] = $requestedBy;
-            $requestedByStatement .= ' OR ua.User = :requestedBy';
-        }
-        $requestedByStatement .= ')';
-    }
+    $joinUsers = $numPlayers < 100;
 
     // Returns an array of the number of players who have achieved each total, up to the max.
     if ($flag === AchievementFlag::OfficialCore) {
-        $countColumn = $hardcore ? 'achievements_unlocked_hardcore' : 'achievements_unlocked';
-        $query = "SELECT pg.$countColumn AS AwardedCount, COUNT(*) AS NumUniquePlayers
-                    FROM player_games AS pg
-                    $joinStatement
-                    WHERE pg.game_id = :gameId AND pg.$countColumn > 0
-                    $requestedByStatement
-                    GROUP BY AwardedCount
-                    ORDER BY AwardedCount DESC";
+        $countColumn = $isHardcore ? 'player_games.achievements_unlocked_hardcore' : 'player_games.achievements_unlocked_softcore';
+
+        $countQuery = DB::table("player_games")
+            ->selectRaw("$countColumn as AwardedCount, count(*) as NumUniquePlayers")
+            ->where("$countColumn", ">", 0)
+            ->where('player_games.game_id', $gameID)
+            ->groupBy('AwardedCount')
+            ->orderByDesc('AwardedCount');
+
+        if ($joinUsers) {
+            $countQuery->join("useraccounts", "player_games.user_id", "=", "useraccounts.ID")
```

Comment:
nit: All `useraccounts` and `achievements` direct references should be changed to the table names `UserAccounts` and `Achievements`.

Reply:
I'm not sure I understand. Do you mean spelling the table names using capitals (e.g. spelling it `Achievements` instead of `achievements`)?

Reply:
Yes - just to match the existing table names in the database. This may help with a case-sensitive search all across the project when we later rename the tables.

Reply:
Table names updated to use the proper case